internal/v1/handler_compose_image.go

```diff
@@ -180,6 +190,90 @@ func buildRepositories(arch *distribution.Architecture, imageType ImageTypes) []
 	return repositories
 }
 
+func (h *Handlers) buildRepositorySnapshots(ctx echo.Context, arch *distribution.Architecture, imageType ImageTypes, snapshotDate *string) ([]composer.Repository, error) {
+	date, err := time.Parse(time.RFC3339, *snapshotDate)
```

Comment:
Nil pointer error when snapshot is `nil`, since it is not required in OpenAPI I think it can be.

Reply:
The check on line 66 will be sufficient, but the signature makes it weird. I'll change this pointer into just a reference.